src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -2566,32 +2566,44 @@ public int ChildCount()
         }
 
         // Determines whether the node supports a compilation / code generation strategy based on walking the node tree.
-        internal bool SupportsCompilation()
+        // Also returns a human-readable string to explain the reason to the source generator.
+        internal bool SupportsCompilation([NotNullWhen(false)] out string? reason)
         {
             if (!StackHelper.TryEnsureSufficientExecutionStack())
             {
-                // If we can't recur further, code generation isn't supported as the tree is too deep.
+                reason = "the regex is too deep";
```

Comment:
```suggestion
                reason = "run-time limits were exceeded";
```

Reply:
Hmm, it checks kind of the same thing with the other "the regex is too deep" check, but looks like this one usually fails on much deeper regexes. Makes sense to add a different message.

Reply:
> it checks kind of the same thing with the other "the regex is too deep" check, but looks like this one usually fails on much deeper regexes

One of them is checking to see if we're too deep on the stack to make what the runtime estimates is an average method call.  The other is checking to see if the node tree is so deep that we'd likely produce C# code that would fail to compile because of C# compiler limits it puts in place; those limits aren't documented, however, so we just chose a depth that's so large it accommodates pretty much any regex someone would write but small enough that we shouldn't hit those limits.